### PR TITLE
Protect object property from being overridden

### PR DESCRIPTION
**Pull request by heypnus**

### CODE REVIEW

File: controllers/antreainstall_controller.go [hunk 1]
```diff
@@ -240,6 +240,42 @@ func (r *AntreaInstallReconciler) Reconcile(request ctrl.Request) (ctrl.Result,
...
+	log.Info(fmt.Sprintf("reconciling %s", objDesc))
+
+	// Get existing
+	existing := &uns.Unstructured{}
+	existing.SetGroupVersionKind(gvk)
+	err := c.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, existing)
+
+	if err != nil && apierrors.IsNotFound(err) {
+		log.Info(fmt.Sprintf("does not exist, creating %s", objDesc))
+		err := c.Create(ctx, obj)
```
Comment (jwsui) [inline controllers/antreainstall_controller.go:261]: Maybe I prefer `if err := xxx; err != nil {...}`

File: controllers/antreainstall_controller.go [hunk 2]
```diff
@@ -240,6 +240,42 @@ func (r *AntreaInstallReconciler) Reconcile(request ctrl.Request) (ctrl.Result,
...
+		if err != nil {
+			return err
+		}
+		log.Info(fmt.Sprintf("successfully created %s", objDesc))
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	err = c.Patch(ctx, existing, client.MergeFrom(obj))
```
Comment (jwsui) [inline controllers/antreainstall_controller.go:271]: same here

**Review by jwsui — APPROVED** [review]
lgtm

**Comment by salv-orlando** [comment]
@heypnus  is this PR still valid or can it be closed?

**Comment by heypnus** [comment]
> @heypnus is this PR still valid or can it be closed?

I think it's still valid. Sorry for leaving it there for too long time, will fully test it when I'm available, maybe in 2 weeks.

**Pull request closed without merging**